zipline/pipeline/data/dataset.py

```diff
@@ -18,28 +23,62 @@ class Column(object):
     """
 
     @preprocess(dtype=ensure_dtype)
-    def __init__(self, dtype):
+    def __init__(self, dtype, missing_value=NotSpecified):
         self.dtype = dtype
+        self.missing_value = missing_value
 
     def bind(self, name):
         """
         Bind a `Column` object to its name.
         """
-        return _BoundColumnDescr(dtype=self.dtype, name=name)
+        return _BoundColumnDescr(
+            dtype=self.dtype,
+            missing_value=self.missing_value,
+            name=name,
+        )
 
 
 class _BoundColumnDescr(object):
     """
     Intermediate class that sits on `DataSet` objects and returns memoized
     `BoundColumn` objects when requested.
+
+    This exists so that subclasses of DataSets don't share columns with their
+    parent classes.
     """
-    def __init__(self, dtype, name):
+    def __init__(self, dtype, missing_value, name):
         self.dtype = dtype
+
+        # Calculating missing values here guarantees that we fail quickly if
+        # the user fails to provide a missing value for a dtype that requires
+        # one (e.g. int64), but still enables us to provide an error message
+        # that points to the name of the failing column.
+        if missing_value is NotSpecified:
```

Comment:
an we also check that the missing value is coercable to `dtype` here? For example:

```python
In [17]: class Ds(DataSet):
    c = Column('int64', np.nan)
   ....:     

In [18]: Ds.c
Out[18]: Ds.c::int64
```